lib/jekyll/collection.rb

```diff
@@ -210,12 +210,9 @@ def container
     private
 
     def read_document(full_path)
-      docs << Document.new(full_path, :site => site, :collection => self).tap do |doc|
-        doc.read
-        if !site.publisher.publish?(doc) && site.publisher.hidden_in_the_future?(doc)
-          Jekyll.logger.debug "Skip Publishing:", "#{doc.relative_path} has a future date"
-        end
-      end
+      doc = Document.new(full_path, :site => site, :collection => self)
+      doc.read
+      docs << doc unless doc.data["published"] == false
```

Comment:
These sorts of concerns should be in the Publisher, don’t you think? It’s intention was to be this sort of filter.

Reply:
I didn't want to introduce a new `public_method` in a patch release..

Reply:
I don't see a problem with introducing a new public method in a patch release.